src/pages/index.js

```diff
@@ -68,10 +102,59 @@ editProfileButton.addEventListener("click", () => {
   editProfilePopup.open();
 });
 
+const handleDeleteClick = (card) => {
+  confirmationPopup.open();
+  confirmationPopup.setSubmitCallback(() => {
+    confirmationPopup.setButtonText(true, "Deleting...");
+
+    api
+      .deleteCard(card._cardId)
+      .then(() => {
+        card.delete();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        confirmationPopup.close();
```

Comment:
* Submit buttons get loading state for always after the first submit https://disk.yandex.ru/i/dN1O2490_BQieg 


You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case.

Reply:
Hey I'm not sure i understand, what exactly is the purpose of the finally?

Reply:
> Hey I'm not sure i understand, what exactly is the purpose of the finally?

`finally` is for the code that must work after any result: `sucess` or `failure`